samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -52,6 +52,26 @@ void registerNextOperator(OperatorImpl<RM, ?> nextOperator) {
   public abstract void onNext(M message, MessageCollector collector, TaskCoordinator coordinator);
 
   /**
+   * Invoked at every tick. This method delegates to {@link #onTimer(MessageCollector, TaskCoordinator)}
+   *
+   * @param collector
+   * @param coordinator
+   */
+  public final void onTick(MessageCollector collector, TaskCoordinator coordinator) {
+    onTimer(collector, coordinator);
+    propagateTimer(collector, coordinator);
+  }
+
+  /**
+   * Invoked at every tick. Implementations may call {@link #propagateResult} if they wish to propagate the output to registered
```

Comment:
If `propagateResult` is correct here then we should update the documentation at line 77 which states that it should only be called from `onNext`.